selfdrive/controls/lib/events.py

```diff
@@ -0,0 +1,524 @@
+
+AlertSize = log.ControlsState.AlertSize
+AlertStatus = log.ControlsState.AlertStatus
+VisualAlert = car.CarControl.HUDControl.VisualAlert
+AudibleAlert = car.CarControl.HUDControl.AudibleAlert
+EN = car.CarEvent.EventName
+
+# Event types
+class ET:
+  ENABLE = 'enable'
+  PRE_ENABLE = 'preEnable'
+  NO_ENTRY = 'noEntry'
+  WARNING = 'warning'
+  USER_DISABLE = 'userDisable'
+  SOFT_DISABLE = 'softDisable'
+  IMMEDIATE_DISABLE = 'immediateDisable'
+  PERMANENT = 'permanent'
+
+class Events:
+  def __init__(self):
+    self.events = []
+
+  @property
+  def names(self):
+    return self.events
+
+  def add(self, event_name):
+    self.events.append(event_name)
+
+  def any(self, event_types):
+    for e in self.events:
+      for t in event_types:
+        if t in EVENTS[e].keys():
+          return True
+    return False
+
+  def get_events(self, event_types):
+    ret = []
+    for e in self.events:
+      for t in event_types:
+        if t in EVENTS[e].keys():
+          ret.append(e)
+    return ret
+
+  def add_from_msg(self, events):
+    for e in events:
+      self.events.append(e.name.raw)
+
+  def to_msg(self):
+    # TODO: use pycapnp's resizeable list builder
+    ret = []
+    for event_name in self.events:
+      event = car.CarEvent.new_message()
+      event.name = event_name
+      for event_type in EVENTS[event_name].keys():
+        setattr(event, event_type , True)
+      ret.append(event)
+    return ret
+
+class NoEntryAlert(Alert):
+  def __init__(self, alert_text_2, audible_alert=AudibleAlert.chimeError,
+               visual_alert=VisualAlert.none, duration_hud_alert=2.):
+    super().__init__("openpilot Unavailable", alert_text_2, AlertStatus.normal,
+                     AlertSize.mid, Priority.LOW, visual_alert,
+                     audible_alert, .4, duration_hud_alert, 3.)
+
+
+class SoftDisableAlert(Alert):
+  def __init__(self, alert_text_2):
+    super().__init__("TAKE CONTROL IMMEDIATELY", alert_text_2,
+                     AlertStatus.critical, AlertSize.full,
+                     Priority.MID, VisualAlert.steerRequired,
+                     AudibleAlert.chimeWarningRepeat, .1, 2., 2.),
+
+
+class ImmediateDisableAlert(Alert):
+  def __init__(self, alert_text_2, alert_text_1="TAKE CONTROL IMMEDIATELY"):
+    super().__init__(alert_text_1, alert_text_2,
+                     AlertStatus.critical, AlertSize.full,
+                     Priority.HIGHEST, VisualAlert.steerRequired,
+                     AudibleAlert.chimeWarningRepeat, 2.2, 3., 4.),
+
+
+enable_alert =  Alert(
+                    "",
+                    "",
+                    AlertStatus.normal, AlertSize.none,
+                    Priority.MID, VisualAlert.none, AudibleAlert.chimeEngage, .2, 0., 0.)
+
+disable_alert = Alert(
+                    "",
+                    "",
+                    AlertStatus.normal, AlertSize.none,
+                    Priority.MID, VisualAlert.none, AudibleAlert.chimeDisengage, .2, 0., 0.)
+
+EVENTS = {
```

Comment:
Would it be better to put EVENTS to a seperate file? it may be very helpful for multilanguage support

Reply:
I'd prefer it'd stay here for now. When multilanguage gets added, it'll be easy to pull out